app/screens/AppNavigator/screens/Dex/PoolSwap/PoolSwapScreen.tsx

```diff
@@ -188,21 +255,62 @@ function SwapSummary ({ poolpair, tokenA, tokenB }: SwapSummaryItems): JSX.Eleme
   return (
     <View style={tailwind('mt-4')}>
       <PriceRow
+        testID='price'
         title={translate('screens/PoolSwapScreen', 'Price')}
         values={price}
       />
       <PriceRow
+        testID='estimated'
         title={translate('screens/PoolSwapScreen', 'Estimated to receive')}
-        values={[{ amount: '1', symbol: tokenB.symbol }]}
+        values={[{ amount: new BigNumber(tokenAAmount).times(price[1].amount).toFixed(), symbol: tokenB.symbol }]}
       />
       <PriceRow
+        testID='minimum'
         title={translate('screens/PoolSwapScreen', 'Minimum to receive')}
-        values={[{ amount: '1', symbol: tokenB.symbol }]}
+        values={[{ amount: tokenBAmount, symbol: tokenB.symbol }]}
       />
       <PriceRow
+        testID='fee'
         title={translate('screens/PoolSwapScreen', 'Liquidity provider fee')}
-        values={[{ amount: '1', symbol: 'DFI' }]}
+        values={[{ amount: '0.001', symbol: 'DFI' }]}
       />
     </View>
   )
 }
+
+interface DexForm {
+  fromToken: DerivedTokenState
+  toToken: DerivedTokenState
+  fromAmount: BigNumber
+  toAmount: BigNumber
+}
+async function constructSignedSwapAndSend (
+  account: WhaleWalletAccount, // must be both owner and recipient for simplicity
+  dexForm: DexForm,
+  // dispatch: Dispatch<any>
+  whaleAPI: WhaleApiClient
+): Promise<void> {
+  const builder = account.withTransactionBuilder()
+
+  const maxPrice = dexForm.fromAmount.div(dexForm.toAmount)
+
+  // will be handled in jellyfish soon (submit maxPrice as a single BN)
+  const integer = maxPrice.integerValue(BigNumber.ROUND_FLOOR)
+  const fraction = maxPrice.modulo(1).times('1e8').integerValue(BigNumber.ROUND_FLOOR)
```

Comment:
@fuxingloh please check this, should be handled automatically in jellyfish after Jeremy's PR merged.

Reply:
The `maxPrice` format is quite weird, once txn_builder is out we can be more certain about it's use cases and edge cases.